stackslib/src/chainstate/burn/operations/leader_block_commit.rs

```diff
@@ -3032,6 +3147,293 @@ mod tests {
+            network_name: "testnet".to_string(),
+            working_dir: "/nope".to_string(),
+            consensus_hash_lifetime: 24,
+            stable_confirmations: 7,
+            initial_reward_start_block: 0,
+            first_block_height: 0,
+            first_block_timestamp: 0,
+            first_block_hash: BurnchainHeaderHash([0x05; 32]),
+        };
+
+        let default_block_commit = LeaderBlockCommitOp {
+            treatment: vec![],
+            sunset_burn: 0,
+            block_header_hash: BlockHeaderHash([0x22; 32]),
+            new_seed: VRFSeed([0x33; 32]),
+            parent_block_ptr: 125,
+            parent_vtxindex: 0,
+            key_block_ptr: 124,
+            key_vtxindex: 456,
+            memo: vec![0x80],
+            commit_outs: vec![],
+
+            burn_fee: 12345,
+            input: (Txid([0; 32]), 0),
+            apparent_sender: BurnchainSigner::mock_parts(
+                AddressHashMode::SerializeP2PKH,
+                1,
+                vec![StacksPublicKey::from_hex(
+                    "02d8015134d9db8178ac93acbc43170a2f20febba5087a5b0437058765ad5133d0",
+                )
+                .unwrap()],
+            ),
+
+            txid: Txid([0xab; 32]),
+            vtxindex: 444,
+            block_height: 128,
+            burn_parent_modulus: (128 % BURN_BLOCK_MINED_AT_MODULUS) as u8,
+            burn_header_hash: BurnchainHeaderHash([0x11; 32]),
+        };
+
+        let anchor_block_hash = BlockHeaderHash([0xaa; 32]);
+
+        fn reward_addrs(i: usize) -> PoxAddress {
+            let addr = StacksAddress::new(1, Hash160::from_data(&i.to_be_bytes()));
+            PoxAddress::Standard(addr, None)
+        }
+        let burn_addr_0 = PoxAddress::Standard(StacksAddress::burn_address(false), None);
+        let burn_addr_1 = PoxAddress::Standard(StacksAddress::burn_address(true), None);
+        let rs_pox_addrs = RewardSetInfo {
+            anchor_block: anchor_block_hash.clone(),
+            recipients: vec![(reward_addrs(0), 0), (reward_addrs(1), 1)],
+            allow_nakamoto_punishment: true,
+        };
+        let rs_pox_addrs_0b = RewardSetInfo {
+            anchor_block: anchor_block_hash.clone(),
+            recipients: vec![(reward_addrs(0), 0), (burn_addr_0.clone(), 5)],
+            allow_nakamoto_punishment: true,
+        };
+        let rs_pox_addrs_1b = RewardSetInfo {
+            anchor_block: anchor_block_hash.clone(),
+            recipients: vec![(reward_addrs(1), 1), (burn_addr_1.clone(), 5)],
+            allow_nakamoto_punishment: true,
+        };
+
+        fn rev(rs: &RewardSetInfo) -> RewardSetInfo {
+            let mut out = rs.clone();
+            out.recipients.reverse();
+            out
+        }
+
+        fn no_punish(rs: &RewardSetInfo) -> RewardSetInfo {
+            let mut out = rs.clone();
+            out.allow_nakamoto_punishment = false;
+            out
+        }
+
+        let mut test_vectors = vec![
```

Comment:
Per my above comment, I notice in this list of test vectors that you're not covering the case where one PoX address is punished, but another is not.  Regardless of whether or not the values in `.treatment` need to be in the same order as the outputs in `commit_outs`, this case should still be tested.

Reply:
Added in e724d6cf5